### PR TITLE
Cache configuration locally to disk and SharedPreferences for early accessed flags

### PR DESCRIPTION
**Pull request by petzel**
## Description
In order for the SDK to assign users to experiments or check a flag's status, the configuration must be loaded from a remote server. If a flag is evaluated before the configuration is loaded, `null` is returned. This can also result in the same user seeing different experiences for the same flag/experiment, even if no configurations have changed, due to network delays. 

This PR adds multiple levels of caching which will reduce these occurrences. 

#### File Cache
The first level of caching is an encrypted file which stores the entirety of the remote config. The file is loaded asynchronously when the SDK is initialized. If a cached version of the config exists, the `InitializationCallback` will be called only once when the cached version is loaded, even though the response from network will replace the configuration in memory (as well as replace the cache file with the new version). 

#### SharedPreferences Cache
The cached file can still lead to cases where a flag is evaluated as `null` if one is evaluated before the file is read (since it is read asynchronously/off of the main thread). Whenever the SDK detects that a flag was evaluated before the configuration is loaded, it will then save that flag's config to `EncryptedSharedPreferences` (falling back to `SharedPreferences` if necessary) once it is available. Any time a flag evaluation is performed before the entire config is loaded from the disk/network, it will check the `SharedPreferences` for the flag's configuration and use it if one is available.

### CODE REVIEW

File: eppo/src/main/java/cloud/eppo/android/ConfigurationStore.java [hunk 1]
```diff
@@ -1,24 +1,159 @@
...
+                }
+            } catch (Exception e) {
+                Log.e(TAG, "Unable to cache config to file", e);
+            }
+        });
+    }
+
+    private void deleteConfigCacheFile() {
+        File existingFile = new File(filesDir + File.separator + CACHE_FILE_NAME);
+        if (existingFile.exists()) {
```
Comment (evanmiller) [inline eppo/src/main/java/cloud/eppo/android/ConfigurationStore.java:134]: Suggest `return Files.deleteIfExists(filesDir + File.separator + CACHE_FILE_NAME)`

**Review by evanmiller — APPROVED** [review]
I don't have a lot of context, but seems reasonable!

File: eppo/src/main/java/cloud/eppo/android/EppoHttpClient.java [hunk 2]
```diff
@@ -32,7 +32,7 @@ public void get(String path, RequestCallback callback) {
         HttpUrl httpUrl = HttpUrl.parse(baseUrl + path)
                 .newBuilder()
                 .addQueryParameter("apiKey", apiKey)
-                .addQueryParameter("sdkName", "android")
+                .addQueryParameter("sdkName", "android-testing")
```
Comment (sameerank) [inline eppo/src/main/java/cloud/eppo/android/EppoHttpClient.java:35]: Why the change to the query param?

**Review by sameerank — APPROVED** [review]
